application/core/LSWebUser.php

```diff
@@ -13,6 +13,49 @@ public function __construct()
         $this->loginUrl = Yii::app()->createUrl('admin/authentication', array('sa' => 'login'));
     }
 
+    /**
+     * @inheritDoc
+     * Replace auto getter to check if current user is valid or not
+     */
+    public function getId()
+    {
+        if (empty(parent::getId())) {
+            return parent::getId();
+        }
+        $id = App()->getCurrentUserId();
+        if ($id === 0) {
```

Comment:
Why not `if (empty)` here ?

Reply:
It's the point : function https://github.com/LimeSurvey/LimeSurvey/pull/3525/files#diff-ec864d1f2c65a2e53bd6cc84323c092139025e28b43b107af90ea46f2a5e3ed5R18 check session.

If sessio is set : check user : if user is invalid : return 0. User was still connected but are disable or deleted.

Reply:
What if `App()->getCurrentUserId();` returns `null`?

Reply:
Logically : user is already log out.

See https://github.com/LimeSurvey/LimeSurvey/pull/3525/files#diff-ec864d1f2c65a2e53bd6cc84323c092139025e28b43b107af90ea46f2a5e3ed5R16

> @return int|null user id, **0 mean invalid user**

```
        if ($this->currentUserId && !User::model()->active()->findByPk($this->currentUserId)) {
            $this->currentUserId = 0;
            $this->currentUserId = 0;
        }
        }
```

Maybe we can compare App()->session['loginID'] with [parent::getId]($this->getState('__id');) 

LimeSurvey core use sometime App()->session['loginID'] and sometime App()->user->id …

Maybe i need to extend https://github.com/LimeSurvey/LimeSurvey/blob/03658b26a7b87e4e51cbb432ceeea3517081e74b/vendor/yiisoft/yii/framework/web/auth/CWebUser.php#L288 too ?

Reply:
> What if App()->getCurrentUserId(); returns null?

I mean that maybe we should use empty as it will allow to catch more situations and flexibility.

> Maybe i need to extend ...

Hmm, not sure. Got a bit lost there.
If you need me to join you on the analysis, let's just make a call or something...